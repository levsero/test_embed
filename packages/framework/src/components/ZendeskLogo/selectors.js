import { getZopimId, getLocale } from 'src/redux/modules/base/base-selectors'
import { document, getZendeskHost } from 'utility/globals'
import { i18n } from 'service/i18n'

export const getZendeskLogoLink = (state, linkToChat) => {
  const zopimId = getZopimId(state)
  const locale = getLocale(state)

  if (linkToChat) {
    const params = [
      '?utm_source=webwidgetchat',
      '&utm_medium=poweredbyzendeskchat',
      '&utm_campaign=poweredbyzendesk',
      `&utm_term=${zopimId}`,
      `&utm_content=${getZendeskHost(document)}`,
      `&iref=${zopimId}`,
      `&lang=${locale}`
    ].join('')

    return [i18n.t('embeddable_framework.zendeskLogo.powered_by_url.chat'), params].join('')
  }

  return [
    i18n.t('embeddable_framework.zendeskLogo.powered_by_url.embeddables'),
    '?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image'
  ].join('')
}
