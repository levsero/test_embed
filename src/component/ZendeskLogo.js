import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import { Icon } from 'component/Icon';

export class ZendeskLogo extends Component {
  render() {
    const { fullscreen, rtl, formSuccess, golionLogo, utm } = this.props;
    const logoClasses = classNames({
      'u-linkClean': true,
      'u-posAbsolute u-posEnd--vert': !fullscreen || formSuccess,
      'u-posRelative': fullscreen,
      'u-posStart': !rtl,
      'u-posEnd': rtl,
      [this.props.className]: true
    });
    const logoType = golionLogo ? 'Icon--golion' : 'Icon--zendesk';
    const logoLink = golionLogo && utm === 'ipm' ? 'connect' : 'embeddables'

    const logoUrl = [
      `//www.zendesk.com/${logoLink}/`,
      `?utm_source=${utm}&utm_medium=poweredbyzendesk&utm_campaign=image`
    ].join('');

    return (
      <a href={logoUrl} target='_blank' className={logoClasses}>
        <Icon type={logoType} className='u-posRelative' />
        <span className='u-isHiddenVisually'>zendesk</span>
      </a>
    );
  }
}

ZendeskLogo.propTypes = {
  rtl: PropTypes.bool,
  fullscreen: PropTypes.bool,
  formSuccess: PropTypes.bool,
  className: PropTypes.string,
  utm: PropTypes.string,
  golionLogo: PropTypes.bool
};

ZendeskLogo.defaultProps = {
  rtl: false,
  fullscreen: false,
  formSuccess: false,
  className: '',
  utm: 'webwidget',
  golionLogo: false
};
