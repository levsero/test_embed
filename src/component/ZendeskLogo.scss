@import '../styles/_vars.scss';

.logo {
  composes: u-linkClean from global;
}

.formSuccess {
  position: absolute;
  bottom: #{6 / $font-size}rem;
}

.fullscreen {
  composes: u-posRelative from global;
}

.hidden {
  composes: u-isHiddenVisually from global;
}

.icon {
  composes: u-posRelative from global;
}
