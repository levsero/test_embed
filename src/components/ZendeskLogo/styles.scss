@import 'src/styles/_vars.scss';

.icon {
  position: relative;
  display: inline-block;

  svg {
    min-width: #{50 / $font-size}rem; /* 1 */
    width: #{50 / $font-size}rem;
    height: #{10 / $font-size}rem;

    &:hover,
    &:active,
    &:focus {
      path {
        fill: $zd-color-grey-600;
      }
    }

    path {
      fill: $zd-color-grey-500;
    }
  }
}
