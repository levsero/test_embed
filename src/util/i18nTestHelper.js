import { i18n } from 'service/i18n';
import { createStore } from 'redux';
import reducer from 'src/redux/modules/reducer';
import yaml from 'js-yaml';
import fs from 'fs';
import path from 'path';

i18n.init(createStore(reducer));

// Load fallback translations for tests
const oldVal = global.__DEV__;
const fallbackPath = path.resolve(__dirname, '../../config/locales/translations/embeddable_framework.yml');
const fallbackTranslations = yaml.safeLoad(fs.readFileSync(fallbackPath));

global.__DEV__ = true;
i18n.setFallbackTranslations(fallbackTranslations);
global.__DEV__ = oldVal;

i18n.setLocale('en-US');
