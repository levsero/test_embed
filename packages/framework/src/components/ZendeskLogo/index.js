import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { connect } from 'react-redux'
import { getZendeskLogoLink } from './selectors'
import { TEST_IDS } from 'src/constants/shared'
import LogoIcon from 'icons/widget-icon_zendesk.svg'

import { zdColorGrey600, zdColorGrey500 } from '@zendeskgarden/css-variables'

const Link = styled.a`
  ${(props) => {
    return `
      position: relative;
      display: inline-block;

      svg {
        min-width: ${50 / props.theme.fontSize}rem;
        width: ${50 / props.theme.fontSize}rem;
        height: ${10 / props.theme.fontSize}rem;

        &:hover,
        &:active,
        &:focus {
          path {
            fill: ${zdColorGrey600};
          }
        }

        path {
          fill: ${zdColorGrey500};
        }
      }
  `
  }}
`

const ZendeskLogo = ({ href }) => (
  <Link data-testid={TEST_IDS.ICON_ZENDESK} href={href} target="_blank" tabIndex="-1">
    <LogoIcon title="zendesk" />
  </Link>
)

ZendeskLogo.propTypes = {
  href: PropTypes.string.isRequired,
}

const mapStateToProps = (state, props) => ({
  href: getZendeskLogoLink(state, props.linkToChat),
})

const connectedComponent = connect(mapStateToProps)(ZendeskLogo)

export { connectedComponent as default, ZendeskLogo as Component }
