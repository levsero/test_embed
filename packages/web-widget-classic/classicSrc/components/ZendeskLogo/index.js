import LogoIcon from 'classicSrc/asset/icons/widget-icon_zendesk.svg'
import { TEST_IDS } from 'classicSrc/constants/shared'
import { rgba } from 'polished'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import styled from 'styled-components'
import { zdColorGrey600, zdColorGrey800 } from '@zendeskgarden/css-variables'
import { getZendeskLogoLink } from './selectors'

const Link = styled.a`
  ${(props) => {
    return `
      position: relative;
      display: inline-block;

      &[data-garden-focus-visible] {
        box-shadow: ${props.theme.shadows.md(rgba(props.theme.baseHighlightColor, 0.2))};
        border-radius: 4px;
        background-color: ${props.theme.shadows.md(rgba(props.theme.baseHighlightColor, 0.2))};

        padding: ${4 / props.theme.fontSize}rem;
        margin: ${-4 / props.theme.fontSize}rem;
      }

      svg {
        min-width: ${50 / props.theme.fontSize}rem;
        width: ${50 / props.theme.fontSize}rem;
        height: ${10 / props.theme.fontSize}rem;

        &:hover,
        &:active,
        &:focus {
          path {
            fill: ${zdColorGrey800};
          }
        }

        path {
          fill: ${zdColorGrey600};
        }
      }
    `
  }}
`

const ZendeskLogo = ({ href }) => (
  <Link data-testid={TEST_IDS.ICON_ZENDESK} href={href} target="_blank">
    <LogoIcon title="zendesk" />
  </Link>
)

ZendeskLogo.propTypes = {
  href: PropTypes.string.isRequired,
}

const mapStateToProps = (state, props) => ({
  href: getZendeskLogoLink(state, props.linkToChat),
})

const connectedComponent = connect(mapStateToProps)(ZendeskLogo)

export { connectedComponent as default, ZendeskLogo as Component }
