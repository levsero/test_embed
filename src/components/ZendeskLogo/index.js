import React from 'react'
import PropTypes from 'prop-types'
import { locals as styles } from './styles.scss'
import { connect } from 'react-redux'
import { getZendeskLogoLink } from './selectors'

import LogoIcon from 'icons/widget-icon_zendesk.svg'

const ZendeskLogo = ({ href }) => (
  <a href={href} target="_blank" tabIndex="-1" className={styles.icon}>
    <LogoIcon title="zendesk" />
  </a>
)

ZendeskLogo.propTypes = {
  href: PropTypes.string.isRequired
}

const mapStateToProps = (state, props) => ({
  href: getZendeskLogoLink(state, props.linkToChat)
})

const connectedComponent = connect(mapStateToProps)(ZendeskLogo)

export { connectedComponent as default, ZendeskLogo as Component }
