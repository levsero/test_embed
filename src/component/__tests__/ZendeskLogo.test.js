import 'utility/i18nTestHelper';
import { render } from 'react-testing-library';
import React from 'react';

import { ZendeskLogo } from '../ZendeskLogo';

test('renders the expected classes', () => {
  const { container } = render(<ZendeskLogo />);

  expect(container)
    .toMatchSnapshot();
});

test('does not have the positional classnames when mobile browser is true', () => {
  const { container } = render(<ZendeskLogo fullscreen={true} />);

  expect(container)
    .toMatchSnapshot();
});

test('has the positional classnames for mobile browser and formSuccess is true', () => {
  const { container } = render(<ZendeskLogo formSuccess={true} fullscreen={true} />);

  expect(container)
    .toMatchSnapshot();
});

test('does not has the rtl classnames when rtl language is false', () => {
  const { container } = render(<ZendeskLogo formSuccess={true} rtl={false} />);

  expect(container)
    .toMatchSnapshot();
});

test('has the rtl classnames when rtl language is true', () => {
  const { container } = render(<ZendeskLogo formSuccess={true} rtl={true} />);

  expect(container)
    .toMatchSnapshot();
});

test('has the correct logo when chat is true', () => {
  const { container } = render(<ZendeskLogo logoLink='chat' chatId='123abc' />);

  expect(container)
    .toMatchSnapshot();
});
