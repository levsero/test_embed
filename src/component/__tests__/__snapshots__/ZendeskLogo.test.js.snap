// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not has the rtl classnames when rtl language is false 1`] = `
<div>
  <a
    class="
      logo
      formSuccess
      
      ltr
      
    "
    href="[object Object]?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
    tabindex="-1"
    target="_blank"
  >
    <span
      class="container icon Icon--zendesk"
      type="Icon--zendesk"
    >
      test-file-stub
    </span>
    <span
      class="hidden"
    >
      zendesk
    </span>
  </a>
</div>
`;

exports[`does not have the positional classnames when mobile browser is true 1`] = `
<div>
  <a
    class="
      logo
      
      fullscreen
      ltr
      
    "
    href="[object Object]?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
    tabindex="-1"
    target="_blank"
  >
    <span
      class="container icon Icon--zendesk"
      type="Icon--zendesk"
    >
      test-file-stub
    </span>
    <span
      class="hidden"
    >
      zendesk
    </span>
  </a>
</div>
`;

exports[`has the positional classnames for mobile browser and formSuccess is true 1`] = `
<div>
  <a
    class="
      logo
      formSuccess
      fullscreen
      ltr
      
    "
    href="[object Object]?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
    tabindex="-1"
    target="_blank"
  >
    <span
      class="container icon Icon--zendesk"
      type="Icon--zendesk"
    >
      test-file-stub
    </span>
    <span
      class="hidden"
    >
      zendesk
    </span>
  </a>
</div>
`;

exports[`has the rtl classnames when rtl language is true 1`] = `
<div>
  <a
    class="
      logo
      formSuccess
      
      rtl
      
    "
    href="[object Object]?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
    tabindex="-1"
    target="_blank"
  >
    <span
      class="container icon Icon--zendesk"
      type="Icon--zendesk"
    >
      test-file-stub
    </span>
    <span
      class="hidden"
    >
      zendesk
    </span>
  </a>
</div>
`;

exports[`renders the expected classes 1`] = `
<div>
  <a
    class="
      logo
      formSuccess
      
      ltr
      
    "
    href="[object Object]?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
    tabindex="-1"
    target="_blank"
  >
    <span
      class="container icon Icon--zendesk"
      type="Icon--zendesk"
    >
      test-file-stub
    </span>
    <span
      class="hidden"
    >
      zendesk
    </span>
  </a>
</div>
`;
